components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -15,18 +15,22 @@ use icu_locid::{
 use icu_provider::prelude::*;
 use tinystr::{tinystr4, TinyStr4, TinyStr8};
 
-#[allow(missing_docs)] // TODO(#1027) - Add missing docs.
 /// Used to track the result of a canonicalization operation that potentially modifies its argument in place.
 #[derive(Debug, PartialEq)]
 pub enum CanonicalizationResult {
+    /// The canonicalization operation modified the locale.
     Modified,
+    /// The canonicalization operation did not modify the locale.
     Unmodified,
 }
 
-#[allow(missing_docs)] // TODO(#1027) - Add missing docs.
+/// LocaleCanonicalizer implementation.
```

Comment:
Nit: Since `LocaleCanonicalizer` is the "main" class in this crate, it should have full documentation with more prose and at least one example.

Reply:
Sure, I will do this, but I'm a bit concerned about the amount of duplicated documentation, e.g. between here and what is in `lib.rs` and in the method definitions themselves. There's a maintenance cost associated with maintaining documentation, and a real chance that things will get out of sync.

Reply:
I agree with your sentiment, but I also share @zbraniecki's sentiment that we should err on the side of documenting more things.  Here's how I like to do it.  The module-level docs should contain larger or higher-level, user-focused examples, and the struct-level docs should be a bit shorter and include a smaller example.